frontend/tests/unit/services/web3-provider/injected-provider.spec.ts

```diff
@@ -0,0 +1,103 @@
+import * as providers from '@ethersproject/providers';
+import * as ethersUtils from 'ethers/lib/utils';
+
+import { createInjectedProvider, InjectedProvider } from '@/services/web3-provider';
+import * as utils from '@/services/web3-provider/util-export';
+import { generateChain, getRandomEthereumAddress } from '~/utils/data_generators';
+import { MockedEip1193Provider, MockedWeb3Provider } from '~/utils/mocks/ethereum-provider';
+
+vi.mock('@/services/web3-provider/util-export');
+vi.mock('@ethersproject/providers');
+vi.mock('ethers/lib/utils');
+
+function mockWeb3Provider() {
+  const web3Provider = new MockedWeb3Provider();
+
+  Object.defineProperty(providers, 'Web3Provider', {
+    value: vi.fn().mockImplementation(() => web3Provider),
+  });
+
+  return web3Provider;
+}
+
+describe('injected-provider', () => {
+  beforeEach(() => {
+    mockWeb3Provider();
+  });
+
+  describe('createInjectedProvider()', () => {
+    it('creates & initializes an injected wallet provider', async () => {
+      Object.defineProperty(utils, 'detectEthereumProvider', {
+        value: vi.fn().mockResolvedValue(new MockedEip1193Provider()),
+      });
+
+      const chain = generateChain({ identifier: 2 });
+      const signerAddress = getRandomEthereumAddress();
+      const signer = 'fake-signer';
+
+      const web3Provider = mockWeb3Provider();
+      web3Provider.getNetwork = vi.fn().mockReturnValue({ chainId: chain.identifier });
+      web3Provider.listAccounts = vi.fn().mockReturnValue([signerAddress]);
+      web3Provider.getSigner = vi.fn().mockReturnValue(signer);
+
+      const injectedProvider = await createInjectedProvider();
+
+      expect(injectedProvider?.chainId.value).toBe(chain.identifier);
+      expect(injectedProvider?.signer.value).toBe(signer);
+      expect(injectedProvider?.signerAddress.value).toBe(signerAddress);
+    });
+    describe('when no injected provider is available', () => {
+      it('returns undefined', async () => {
+        Object.defineProperty(utils, 'detectEthereumProvider', {
+          value: vi.fn().mockResolvedValue(null),
+        });
+
+        const injectedProvider = await createInjectedProvider();
+
+        expect(injectedProvider).toBeUndefined();
+      });
+    });
+  });
+
+  describe('InjectedProvider', () => {
+    describe('requestSigner()', () => {
```

Comment:
Since the method moved to InjectedProvider, I think it is better to remove the respective tests from the MetaMask provider. Would mean we would have to maintain it in two places, if something changes.

Reply:
Good point 👍